x/supply/coinGeckoTypes/coins.go

```diff
@@ -0,0 +1,41 @@
+package coingecko
```

Comment:
I honestly would remove any CoinGecko reference from any file. While it's true that we are currently basing our data fetching on their APIs, it might not be the case in the future. For this reason, I think it's better to rename such references to something more generic. 

Also, I would avoid putting market-related thing inside the `x/supply` folder. Instead, I would create something like `x/pricefeed` which is more generic, putting everything there.